### PR TITLE
Feature/react router

### PR DESCRIPTION
**Pull request by mykyta07**
Link to video🎥:
https://www.awesomescreenshot.com/video/27647626?key=6da9211d988e7354f1758d96ef032ba2

### CODE REVIEW

File: src/pages/Andrii/cards/FormResult.tsx [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+        !textPattern.test(editingData.lastName) ||
+        !textPattern.test(editingData.placeOfIssue) ||
+        !numberPattern.test(editingData.phoneNumber) ||
+        !emailPattern.test(editingData.email)
+      ) {
+        return false;
+      }
+    }
+    return true;
+  };
```
Comment (Emil-Khalus) [inline src/pages/Andrii/cards/FormResult.tsx:95]: please use react-hook-form validation

File: src/pages/Andrii/cards/FormResult.tsx [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+                                value={editingData.firstName}
+                                error={editingData.firstName === '' || !textPattern.test(editingData.firstName)}
+                                helperText={editingData.firstName === '' ? 'This field is required' : !textPattern.test(editingData.firstName) ? 'First Name must start with a capital letter and cannot contain numbers or special characters' : ''}
+                                onChange={handleInputChange('firstName')}
+                              />
+                              <StyledTextField
+                                value={editingData?.lastName || ''}
+                                error={editingData.lastName === '' || !textPattern.test(editingData.lastName)}
+                                helperText={editingData.lastName === '' ? 'This field is required' : !textPattern.test(editingData.lastName) ? 'Last Name must start with a capital letter and cannot contain numbers or special characters' : ''}
+                                onChange={handleInputChange('lastName')}
```
Comment (Emil-Khalus) [inline src/pages/Andrii/cards/FormResult.tsx:178]: you can use react-hook-form for tracking input value and errors state

File: src/pages/Layout.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import { Link, Outlet } from 'react-router-dom';
+import {AppBar, Toolbar, Typography, Button } from '@mui/material';
+
+function Layout() {
+
+    // const theme = useTheme();
```
Comment (Emil-Khalus) [inline src/pages/Layout.tsx:6]: drop comment please

File: src/pages/Andrii/cards/FormResult.tsx [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+      [prop]: event.target.value,
+    }) as RentCar);
+  };
+
+  const handleDateChange = (prop: keyof RentCar) => (newValue: Dayjs | null) => {
+    setEditingData(prev => ({
+      ...prev,
+      [prop]: newValue,
+    }) as RentCar);
+  };
```
Comment (Emil-Khalus) [inline src/pages/Andrii/cards/FormResult.tsx:73]: you can replace it with react-hook-form

File: src/pages/Andrii/cards/FormResultMobile.tsx [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+          !textPattern.test(editingData.lastName) ||
+          !textPattern.test(editingData.placeOfIssue) ||
+          !numberPattern.test(editingData.phoneNumber) ||
+          !emailPattern.test(editingData.email)
+        ) {
+          return false;
+        }
+      }
+      return true;
+    };
```
Comment (Emil-Khalus) [inline src/pages/Andrii/cards/FormResultMobile.tsx:62]: the same here

**Review by Emil-Khalus — CHANGES_REQUESTED** [review]

File: src/pages/Layout.tsx (hunk 3 above)
Comment (Thundem) [reply src/pages/Layout.tsx]: i fixed it

**Comment by Emil-Khalus** [comment]
I think this fields should be cleared after submit
image

File: src/pages/Andrii/cards/FormResult.tsx [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+import { StyledButtonDelete } from '../styledComponents/StyledButtonDelete'
+import { StyledButtonEdit } from '../styledComponents/StyledButtonEdit'
+import { StyledButtonSave } from '../styledComponents/StyledButtonSave'
+import { StyledTextField } from '../styledComponents/StyledTextField'
+import { createTransform } from '../animations/animation' 
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DateTimePicker } from '@mui/x-date-pickers/DateTimePicker';
+import { DateTimeValidationError } from '@mui/x-date-pickers/models';
+// import { DragDropContext, Droppable, Draggable } from 'react-beautiful-dnd';
```
Comment (Emil-Khalus) [inline src/pages/Andrii/cards/FormResult.tsx:16]: comment

File: src/pages/Andrii/forms/RentCarForm.tsx [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+        width: '100%', 
+        padding: '10px',
+        boxSizing: 'border-box',
+    }
+
+    const getError = (fieldName: keyof RentCar) => ({
+        borderColor: errors[fieldName] ? "error.main" : '#bcbcbc',
+        '& .MuiOutlinedInput-root': {
+            '& fieldset': {
+                borderColor: errors[fieldName] ? "error.main" : '#bcbcbc',
```
Comment (Emil-Khalus) [inline src/pages/Andrii/forms/RentCarForm.tsx:75]: hardcoded color values, please replace with colors from theme

File: src/pages/Car.tsx [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+  const { control } = useForm<CarGroup>();
+  const { fields, append, update, remove } = useFieldArray({
+    control,
+    name: "cars",
+  });
+
+  const [editingCardIndex, setEditingCardIndex] = useState<number | null>(null);
+
+  const handleEditClick = (index: number) => {
+    // console.log("EDITING INDEX", index);
```
Comment (Emil-Khalus) [inline src/pages/Car.tsx:18]: comment

File: src/pages/Layout.tsx [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    const NavBar = {
+        width: '100%', 
+        backgroundColor: '#c19f7b', 
+        boxShadow: '0 0 7px rgba(0,0,0,0.15)',
+    }
+
+    const Buttons = {
+        '&:hover': {
+            backgroundColor: '#d3c5b6',
+            color: '#5d3f1f',
```
Comment (Emil-Khalus) [inline src/pages/Layout.tsx:17]: hardcoded colors

**Review by Emil-Khalus — APPROVED** [review]

**Pull request merged**